### PR TITLE
Homme(xx)/SL: Add SL-transport feature to doubly periodic mode.

### PR DESCRIPTION
**Pull request by ambrad**
* Add semi-Lagrangian transport feature to planar doubly periodic mode.
* Add a BFB test for F90/C++ dycore consistency.
* Fix some C++ dycore issues related to qsize=0.
* Add a bit of infrastructure to make running standalone tests on Summit much easier.

[BFB] except for new tests in homme_integration

### CODE REVIEW

**Comment by ambrad** [comment]
Testing:
* Chrysalis homme_integration: to do
* Chrysalis e3sm_developer: to do
* BFB tests on Ascent/Summit: pass

File: components/homme/src/share/cxx/Hommexx_config.h.in [hunk 1]
```diff
@@ -1,9 +1,6 @@
 #ifndef HOMMEXX_CONFIG_H
 #define HOMMEXX_CONFIG_H
 
-// Identifies whether this is a cuda/hip build
-#cmakedefine HOMMEXX_ENABLE_GPU
```
Comment (bartgol) [inline components/homme/src/share/cxx/Hommexx_config.h.in:5]: Is this removal intended? There seems to be plenty of code that depends on this macro being defined/undefined.

File: components/homme/src/share/sl_advection.F90 [hunk 2]
```diff
@@ -937,9 +937,9 @@ subroutine calc_vertically_lagrangian_levels( &
 
     dp_neg_min = reconstruct_and_limit_dp(elem%state%dp3d(:,:,:,tl%np1), &
          dt, dp_tol, eta_dot_dpdn(:,:,:,1), dprecon)
-#ifndef NDEBUG
+#if 0
```
Comment (bartgol) [inline components/homme/src/share/sl_advection.F90:940]: Why not leaving the existing ifndef?

File: components/homme/src/theta-l_kokkos/cxx/cxx_f90_interface_theta.cpp [hunk 3]
```diff
@@ -581,17 +581,19 @@ void init_boundary_exchanges_c ()
     bmm[MPI_EXCHANGE_MIN_MAX]->set_connectivity(connectivity);
   }
 
-  if (params.transport_alg == 0) {
-    // Euler BEs
-    auto& esf = c.get<EulerStepFunctor>();
-    esf.reset(params);
-    esf.init_boundary_exchanges();
-  } else {
+  if (params.qsize > 0) {
```
Comment (bartgol) [inline components/homme/src/theta-l_kokkos/cxx/cxx_f90_interface_theta.cpp:584]: Does theta formulation make any sense without qv?

File: components/homme/src/share/cxx/Hommexx_config.h.in (hunk 1 above)
Comment (ambrad) [reply components/homme/src/share/cxx/Hommexx_config.h.in:5]: This symbol was being defined in both Hommexx_config.h and config.h. The distinction between these two is as follows, as I understand things. The first is for symbols used only in the Hommexx code. The second is for symbols used in both Hommexx and Homme, the latter including Homme dycores built for BFB testing with Hommexx.

Since the symbol is redundantly defined (leading to a preprocessor warning about this), one should be removed. The symbol is used in Homme as well as Hommexx, so I believe the Hommexx_config.h.in instance should be removed.

My SCREAM test on Ascent, still to do, will check this point.

File: components/homme/src/share/sl_advection.F90 (hunk 2 above)
Comment (ambrad) [reply components/homme/src/share/sl_advection.F90:940]: There was an issue the other day where E3SM's CMake system accidentally removed -DNDEBUG in optimized builds, and that output confused some folks running simulations. It's pretty deep in the weeds. I tend to enable it manually when I want to examine certain things that transport can feel. Since it's not something that benefits others -- and leads to confusion -- I'm removing it from standard builds.

File: components/homme/src/theta-l_kokkos/cxx/cxx_f90_interface_theta.cpp (hunk 3 above)
Comment (ambrad) [reply components/homme/src/theta-l_kokkos/cxx/cxx_f90_interface_theta.cpp:584]: Yes, it can run 'dry'.

File: components/homme/src/share/cxx/Hommexx_config.h.in (hunk 1 above)
Comment (ambrad) [reply components/homme/src/share/cxx/Hommexx_config.h.in:5]: Ok, I think I see why this symbol got redundantly defined. The SCREAM build indeed is not seeing the symbol. I'm going to investigate to see if I can do something about that. In the worst case I'll revert this commit, since redundant defines are not pretty but also not fatal.

**Review by oksanaguba — APPROVED** [review]

File: components/homme/src/share/cxx/Hommexx_config.h.in (hunk 1 above)
Comment (ambrad) [reply components/homme/src/share/cxx/Hommexx_config.h.in:5]: I think I have a way to do this to avoid redefining this symbol. Will test tomorrow.

Comment (ambrad) [reply components/homme/src/share/cxx/Hommexx_config.h.in:5]: The last commit has resolved this both issues: working in SCREAM and avoiding redundant defines in standalone HOMMEXX.

**Review by bartgol — APPROVED** [review]

**Comment by ambrad** [comment]
I reran all the tests noted above after the config.h/Hommexx_config.h resolution, and the status is as reported above. This is ready for integration, but I know `next` is backed up right now, so I'll likely hold off for a week before attempting to find a day to integrate this PR.
